.github/workflows/scripts/comments.mjs

```diff
@@ -75,6 +77,59 @@ export async function commentDiff (
   }
 }
 
+/**
+ * Generates comment for stats
+ *
+ * @param {GithubActionContext} githubActionContext
+ * @param {number} issueNumber
+ * @param {StatComment} statComment
+ */
+export async function commentStats (
+  githubActionContext,
+  issueNumber,
+  { titleText, markerText }
+) {
+  const reviewAppURL = getReviewAppUrl(issueNumber)
+
+  // File sizes
+  const fileSizeTitle = '### File sizes'
+  const fileSizes = await getFileSizes()
+  const fileSizeRows = Object.entries(fileSizes).map(([key, value]) => [key, String(value)])
+  const fileSizeHeaders = ['File', 'Size']
+  const fileSizeTable = renderTable(fileSizeHeaders, fileSizeRows)
+  const fileSizeText = [fileSizeTitle, fileSizeTable].join('\n')
+
+  // Modules
+  const modulesTitle = '### Modules'
+  const modules = Object.fromEntries(await Promise.all(modulePaths
+    .filter((modulePath) => typeof modulePath === 'string')
```

Comment:
Stole this from https://github.com/alphagov/govuk-frontend/blob/47db9411a00c49f0d05ead84fb178a97600895d4/packages/govuk-frontend-review/src/app.mjs#L47

But it requires extra guards here to satisfy TypeScript.

Reply:
The extra guards will be because TypeScript knows arrays can be pushed, popped, changed etc

For example, mapping a list of items:

```mjs
list.map((listItem) => [listItem, '123'])
```

By adding `/** @type {const} */` we say "this array is constant, it won't change" (parentheses are key)

```mjs
list.map((listItem) => /** @type {const} */(
  [listItem, '123']
))
```

Which avoids the need for those extra safety checks on later lines

Reply:
I'm a bit confused since [this](https://github.com/alphagov/govuk-frontend/blob/47db9411a00c49f0d05ead84fb178a97600895d4/packages/govuk-frontend-review/src/app.mjs#L47) imports `modulePaths` and maps it to something and `modulePaths` is correctly listed as `string[]`.

If I import `modulePaths` [here](https://github.com/alphagov/govuk-frontend/blob/5804103ad5d417222f3b114c8c2db4e694ba2801/.github/workflows/scripts/comments.mjs#L4), it lists its type as `(string | undefined)[]`, even if all I do is assign it to a const: `const test = modulePaths`.

I was assuming it was something to do with the scope of govuk-frontend-review's `app.mjs` vs a function in the comments script.

Reply:
If you mouse over `modulePath` in the filter params what type does it give you?

Reply:
`modulePaths` is `(string | undefined)[]`.
`modulePath` is `string | undefined`

Reply:
Sorry, you said didn't you

Probably just need that inline comment over in the stats package

Extra checks are caused by `strict: true` in:

https://github.com/alphagov/govuk-frontend/blob/main/.github/workflows/scripts/tsconfig.json

Can turn it off if it's being annoying, or consider replacing with `noImplicitAny` which at least stops "untyped" things from being let through unchecked

Reply:
Ah, tracked it down.

In stats `index.mjs`, [this fella](https://github.com/alphagov/govuk-frontend/blob/5804103ad5d417222f3b114c8c2db4e694ba2801/shared/stats/src/index.mjs#L29-L33) is used as such:

https://github.com/alphagov/govuk-frontend/blob/5804103ad5d417222f3b114c8c2db4e694ba2801/shared/stats/src/index.mjs#L38-L40

and `packageOptions.modulePath` is `modulePath?: string`

Reply:
@domoscargin Nice spot 😊

In that case, we know `modulePath` definitely _isn't_ `undefined`

We can swap this `@type` to `@satisfies` to take the populated values into account too

https://github.com/alphagov/govuk-frontend/blob/5804103ad5d417222f3b114c8c2db4e694ba2801/shared/stats/src/index.mjs#L27

Reply:
You'll still need the `/** @type {const} */` on your `.map()` but can at least drop the `.filter()` now

```mjs
  const modules = Object.fromEntries(await Promise.all(modulePaths
    .map(async (modulePath) => /** @type {const} */(
      [modulePath, await getStats(String(modulePath))]
    ))))
```